src/include/miopen/solver.hpp

```diff
@@ -4666,17 +4666,18 @@ struct ConvHipImplicitGemmFwdXdlops final
 struct PerformanceConfigHipImplicitGemmBwdXdlops
     : PerfConfigBase<PerformanceConfigHipImplicitGemmBwdXdlops>
 {
-    int index;
-    std::string kernel_id;
+    int index             = 0;
+    std::string kernel_id = "";
     std::vector<std::string> valid_kernels;
+
```

Comment:
[Q] AFAICS we use `kernel_id` as PerformanceConfig value, and we still need `index` and `valid_kernels` for traversing the ComputedContainer. Maybe we can get rid of `kernel_id` and store only `index` in the database? The positive side effect is smalled perf-db footprint.

Or this would make PerformanceConfig incompatible between CK versions because `valid_kernels` may become different?

Reply:
I think we can't since we do not have any guarantees that the order of the kernels is persistent. And it may break backward compatibility as well.

Reply:
[Recommendation]
@CAHEK7 Please put C++ comment that summarizes this review thread, i.e. explains why we need string `kernel_id` as well as `index` and `valid_kernels` in the CK's PerformanceConfigs.